### PR TITLE
feat(cli): install-aware ao update, startup notifier, doctor version check

### LINKED ISSUE #1136: feat: improve update experience for all install methods
## Problem

`ao update` only works for git source installs. For npm global installs (the recommended install path), it fails with a confusing "The update command must run inside the Agent Orchestrator git checkout" error. There is no startup version check, no update notification, and no mechanism for npm users to know when a new version is available.

### Current behavior

- **npm users:** `ao update` fails because `scripts/ao-update.sh` assumes a git repo. Users must independently discover new versions and run `npm install -g @aoagents/ao@latest` themselves.
- **Source users:** `ao update` works (git pull + pnpm rebuild + npm link), but only if on `main` with a clean working tree.
- **No notifications:** No startup version check, no dashboard banner, no `ao doctor` version freshness check.

### Root cause

`script-runner.ts:6` resolves `DEFAULT_REPO_ROOT` by walking up 4 directories from the CLI package. For npm global installs this lands inside `node_modules`, where the git check at `ao-update.sh:97` fails. The update command delegates blindly to the shell script without detecting install method first.

### Existing infrastructure

Install method detection partially exists already:
- `isInstalledUnderNodeModules()` in `packages/cli/src/lib/dashboard-rebuild.ts` distinguishes npm vs source
- `preflight.ts` already branches behavior (error messages, fix hints) based on install method
- `getCliVersion()` in `packages/cli/src/options/version.ts` reads current version from package.json

### 1. Shared update service (`packages/cli/src/lib/update-check.ts`)

Single module owning all version-check logic:
- `detectInstallMethod()` — reuses `isInstalledUnderNodeModules()`, checks for git repo + `scripts/ao-update.sh`
- `checkForUpdate()` — hits npm registry (`registry.npmjs.org/@aoagents%2Fao/latest`) with 3s timeout, caches result for 24h at `~/.cache/ao/update-check.json`
- `getUpdateCommand(method)` — returns the right upgrade command per install method

### 2. Startup version notifier (`packages/cli/src/index.ts`)

Non-blocking, cache-only check before command execution. Prints to stderr if outdated:

```
Update available: 0.2.2 -> 0.3.0 — Run: ao update
```

Skipped in CI, non-TTY, and meta commands (`ao update`, `ao doctor`, `ao --version`). Background refresh kicks off a fetch for next run.

### 3. Install-aware `ao update` (`packages/cli/src/commands/update.ts`)

Route by install method in TypeScript before touching shell scripts:
- **git:** existing `ao-update.sh` flow (unchanged)
- **npm-global:** run `npm install -g @aoagents/ao@latest` (with confirmation)
- **unknown:** print version info + suggest manual upgrade
- **New flag:** `--check` prints JSON version info without upgrading

### 4. Doctor version freshness check (`packages/cli/src/commands/doctor.ts`)

Add a WARN-level check (not FAIL) showing current vs latest version:
```
WARN ao v0.2.2 is outdated (latest: v0.3.0). Run: ao update
```

### 5. Tests (`packages/cli/src/__tests__/lib/update-check.test.ts`)

Cover install detection, cache read/write/staleness, registry fetch (mocked), and command routing.

## Implementation order

1. Shared update service (foundation)
2. Startup notifier (biggest user-facing value)
3. Install-aware `ao update` (fixes the broken command)
4. Doctor version check (natural fit)
5. Tests

## Cross-model validation

This approach was independently validated by OpenAI Codex, which agreed on the architecture, implementation order, and flagged the same pitfalls (don't block startup on network, don't auto-install silently, v1 of npm-mode should confirm before executing).

### PR DESCRIPTION
**Pull request by ashish921998**
## Summary

- **Install-aware `ao update`**: Detects install method (git source, npm global, pnpm global) and routes accordingly. Git installs use existing shell script. npm/pnpm installs prompt to run the correct package manager command. Unknown installs print guidance. New `--check` flag outputs JSON version info.
- **Startup version notifier**: Reads cached update check synchronously before `parseAsync()` (no network on startup). Prints one-liner to stderr when outdated. Background cache refresh runs after command completes via unref'd timer.
- **Doctor version check**: Shows WARN when cached version is outdated, PASS when current. Cache-only, no network dependency in diagnostics.
- **Shared update service** (`update-check.ts`): Install detection via `import.meta.url` with npm/pnpm/git/local discrimination, npm registry fetch with 3s timeout, 24h cache with manual-upgrade detection via `currentVersionAtCheck` field, pre-release version handling.

Closes #1136

## Test Coverage

406 tests pass (24 test files). 99.3% diff coverage (2 uncovered lines: defensive `getCliVersion()` fallback).

Key test areas:
- `update-check.test.ts` (73 tests): isVersionOutdated (7 semver cases + pre-release), classifyInstallPath (10 paths including npm/pnpm/nvm/local), cache lifecycle (8 states), registry fetch (9 failure modes), maybeShowUpdateNotice (13 including all skipArgs + env vars), checkForUpdate (7 including cache bypass/write/no-write)
- `update.test.ts` (23 tests): git/npm-global/pnpm-global/unknown routing, --check JSON output, npm install confirm/decline/spawn-failure/non-TTY, flag warnings
- `doctor.test.ts` (+3 tests): version freshness PASS/WARN/no-cache
- `index.test.ts` (3 tests): parseAsync + ConfigNotFoundError + re-throw

## Pre-Landing Review

No issues found. Clean on all critical categories (SQL, race conditions, shell injection, enum completeness). Two informational issues auto-fixed during development (unnecessary type cast, unreachable EACCES handler).

## Scope Drift

Scope Check: CLEAN
Intent: Install-aware ao update, startup version notifier, doctor version check
Delivered: Exactly that, plus pnpm global support and comprehensive test coverage

## Plan Completion

Plan file: `~/.claude/plans/piped-inventing-zebra.md`
All items DONE or CHANGED. Phase 1 (install-aware update) and Phase 2 (notifier + doctor) both complete.

## Test plan

- [x] `pnpm --filter @aoagents/ao-cli typecheck` passes
- [x] 406/406 CLI tests pass (24 test files)
- [x] CI lint passes
- [x] Multiple Codex reviews conducted (3 rounds, all findings addressed)
- [x] Human code review feedback addressed (itrytoohard + Cursor Bugbot)
- [ ] Manual: `ao update` from git checkout runs existing shell script
- [ ] Manual: `ao update --check` prints JSON
- [ ] Manual: `ao doctor` shows version line

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by i-trytoohard — APPROVED** [review]

**Comment by i-trytoohard** [comment]
@cursor please review this PR thoroughly, and flag any new code that isn't covered in tests that could have been

Be more rigorous in reviewing the unit tests & integration tests, makeing sure that cover every scenario that I user could run into

File: packages/cli/src/lib/update-check.ts [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+// Constants
+// ---------------------------------------------------------------------------
+
+const REGISTRY_URL = "https://registry.npmjs.org/@aoagents%2Fao/latest";
+const FETCH_TIMEOUT_MS = 3000;
+const CACHE_TTL_MS = 24 * 60 * 60 * 1000; // 24 hours
+
+// ---------------------------------------------------------------------------
+// Install detection
+// ---------------------------------------------------------------------------
```
Comment (i-trytoohard) [inline packages/cli/src/lib/update-check.ts:48]: 🔴 **Untested branch:** The `return "npm-global"` path (when `import.meta.url` contains `/node_modules/`) has **zero test coverage**. This is one of three possible return values for install detection — pretty critical.

Consider mocking `import.meta.url` or extracting the path check into a testable function.

File: packages/cli/src/lib/update-check.ts [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+
+  if (latestVersion) {
+    writeCache({
+      latestVersion,
+      checkedAt: now,
+      currentVersionAtCheck: currentVersion,
+    });
+  }
+
+  return {
```
Comment (i-trytoohard) [inline packages/cli/src/lib/update-check.ts:256]: 🔴 **Untested:** The actual notice-printing happy path (stderr IS TTY, no env overrides, no skip args, cache shows outdated) is **never tested**. You test three guard clauses but not the actual output.

Missing:
- `maybeShowUpdateNotice` printing when outdated
- `AGENT_ORCHESTRATOR_CI` env var guard
- Skip args (`update`, `doctor`, `--version`, etc.)
- `scheduleBackgroundRefresh` — no test at all

File: packages/cli/src/commands/update.ts [hunk 3]
```diff
@@ -1,28 +1,161 @@
...
+// ---------------------------------------------------------------------------
+// npm-global install
+// ---------------------------------------------------------------------------
+
+async function handleNpmUpdate(opts: {
+  skipSmoke?: boolean;
+  smokeOnly?: boolean;
+}): Promise<void> {
+  if (opts.skipSmoke || opts.smokeOnly) {
+    console.log(
```
Comment (i-trytoohard) [inline packages/cli/src/commands/update.ts:92]: 🔴 **Major coverage gap in `handleNpmUpdate`:** The following branches have **no tests**:

1. **User confirms prompt** (`promptConfirm → true`) → triggers `runNpmInstall` + `invalidateCache` + success message
2. **`runNpmInstall` non-zero exit** → chalk warning with sudo suggestion (line 110-116)
3. **`runNpmInstall` spawn error** → promise rejection from `child.on('error')`
4. **`--skip-smoke` / `--smoke-only` with npm-global** → yellow warning about git-only flags
5. **Non-TTY environment** (`IS_TTY=false`) → prints `Run: <command>` without prompting
6. **`promptConfirm` throws** → unhandled rejection

The npm-global path is the most complex flow in this PR and it's essentially untested end-to-end.

File: packages/cli/src/commands/update.ts [hunk 4]
```diff
@@ -1,28 +1,161 @@
...
+    );
+  }
 
-      await executeScriptCommand("ao-update.sh", args);
+  const info = await checkForUpdate({ force: true });
+
+  if (!info.latestVersion) {
+    console.error(chalk.red("Could not reach npm registry. Check your network and try again."));
+    process.exit(1);
+  }
```
Comment (i-trytoohard) [inline packages/cli/src/commands/update.ts:102]: ⚠️ **Untested edge case:** When `info.latestVersion` is `null` (registry unreachable) and `info.isOutdated` is `false`, the flow falls through to `!info.isOutdated` and prints "Already on latest version" — which is misleading when the registry couldn't be reached. The `handleNpmUpdate` path for `latestVersion === null` should be tested separately from the "genuinely up to date" case.

Consider: if `!info.latestVersion` on line 96, show the network error instead of falling through.

File: packages/cli/src/commands/doctor.ts [hunk 5]
```diff
@@ -377,6 +378,42 @@ async function sendTestNotifications(
...
+
+function checkVersionFreshness(): void {
+  console.log("");
+  console.log("Version:");
+
+  const current = getCurrentVersion();
+  const cached = readCachedUpdateInfo();
+
+  if (!cached) {
+    pass(`ao v${current} installed (run any ao command to check for updates)`);
```
Comment (i-trytoohard) [inline packages/cli/src/commands/doctor.ts:393]: ⚠️ **Code smell:** This inline semver comparison is a **duplicate** of the private `isVersionOutdated()` in `update-check.ts`. Two different implementations of the same logic = drift risk.

Suggest exporting `isVersionOutdated` from `update-check.ts` and calling it here. Also, neither implementation handles pre-release tags (e.g. `0.2.2-beta.1`) — `Number('beta')` → `NaN`, which breaks the comparison.

File: packages/cli/src/lib/update-check.ts [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+
+/** Read cached update info. Returns null if missing, expired, corrupt, or version-mismatched. */
+export function readCachedUpdateInfo(): CacheData | null {
+  try {
+    const raw = readFileSync(getCachePath(), "utf-8");
+    const data = JSON.parse(raw) as CacheData;
+
+    if (!data.latestVersion || !data.checkedAt) return null;
+
+    // Cache is stale if user upgraded since the check
```
Comment (i-trytoohard) [inline packages/cli/src/lib/update-check.ts:166]: 💡 `getCacheDir()` reads `XDG_CACHE_HOME` but this override is **never tested**. If someone sets `XDG_CACHE_HOME=/tmp`, the cache path changes — worth a unit test to verify.

File: packages/cli/src/lib/update-check.ts [hunk 7]
```diff
@@ -0,0 +1,279 @@
...
+    mkdirSync(dir, { recursive: true });
+    writeFileSync(getCachePath(), JSON.stringify(data, null, 2));
+  } catch {
+    // Best-effort — don't crash if cache dir is unwritable
+  }
+}
+
+export function invalidateCache(): void {
+  try {
+    unlinkSync(getCachePath());
```
Comment (i-trytoohard) [inline packages/cli/src/lib/update-check.ts:194]: 💡 `writeCache` swallows all errors silently. Not a bug, but worth a test to verify it **doesn't throw** when the cache dir is unwritable (e.g. permissions error).

File: packages/cli/src/commands/update.ts [hunk 8]
```diff
@@ -1,28 +1,161 @@
...
+  detectInstallMethod,
+  getCurrentVersion,
+  getUpdateCommand,
+  invalidateCache,
+} from "../lib/update-check.js";
+import { promptConfirm } from "../lib/prompts.js";
+
+const IS_TTY = Boolean(process.stdin.isTTY && process.stdout.isTTY);
 
 export function registerUpdate(program: Command): void {
```
Comment (i-trytoohard) [inline packages/cli/src/commands/update.ts:19]: ⚠️ **Testability issue:** `IS_TTY` is captured at module load time as a module-level constant. This makes it impossible to test the non-TTY branch in `handleNpmUpdate` without mocking `process.stdin.isTTY` *before* module import. Consider making it a function call or checking inline.

**Review by i-trytoohard — CHANGES_REQUESTED** [review]
## Code Review — PR #1156

**Overall:** Solid feature work. The update-check service is well-structured with good separation of concerns. Core library has 27 tests covering the main paths. But there are significant coverage gaps in the **npm-global update flow** (the most complex path) and the **startup notifier**.

### 🔴 Must-Fix (Missing Tests)
1. **`handleNpmUpdate` end-to-end** — 6 branches untested (user confirms, spawn failure, spawn error, non-TTY, flag warnings, prompt throws). This is the most complex flow in the PR.
2. **`maybeShowUpdateNotice` happy path** — The actual notice printing is never tested, only guard clauses.
3. **`detectInstallMethod` npm-global branch** — One of three possible return values, zero coverage.
4. **`AGENT_ORCHESTRATOR_CI` env guard** — Checked in code, no test.

### ⚠️ Should-Fix
5. **Duplicate semver comparison** — `doctor.ts` has its own inline version instead of using the one in `update-check.ts`. Drift risk. Neither handles pre-release tags.
6. **`scheduleBackgroundRefresh`** — No test at all.
7. **`IS_TTY` module-level constant** — Captured at import time, makes non-TTY branch untestable.
8. **npm registry unreachable misleading message** — Falls through to "Already on latest" when registry is down.

### 💡 Nice-to-Have
9. **`XDG_CACHE_HOME` override** — Untested.
10. **`writeCache` silent failure** — Should verify it doesn't propagate.
11. **`--check` with unreachable registry** — Should still output valid JSON.
12. **`isVersionOutdated` edge cases** — Equal versions, current > latest, non-standard version strings.

### ✅ Looks Good
- Clean type definitions (`InstallMethod`, `UpdateInfo`, `CacheData`)
- Cache invalidation on version change (smart `currentVersionAtCheck` field)
- 24h TTL with manual-upgrade detection
- Synchronous startup read (no network) + background refresh via `unref()`
- Good error handling in `fetchLatestVersion` (timeout, non-ok, JSON parse, missing field)
- Existing git install flow preserved and working

**Verdict: Changes Requested** — the npm-global update path is the marquee feature of this PR and it's effectively untested.

File: packages/cli/src/commands/update.ts [hunk 9]
```diff
@@ -1,28 +1,167 @@
...
+    console.log(
+      chalk.yellow("--skip-smoke and --smoke-only only apply to git source installs. Ignoring."),
+    );
+  }
+
+  const info = await checkForUpdate({ force: true });
+
+  if (!info.latestVersion) {
+    console.error(chalk.red("Could not reach npm registry. Check your network and try again."));
+    process.exit(1);
```
Comment (i-trytoohard) [inline packages/cli/src/commands/update.ts:101]: ⚠️ **Non-TTY exits with code 1:** When not a TTY, the flow prints the command then `process.exit(1)`. This means scripts checking exit codes will see failure even though nothing went wrong — the user just needs to run a command manually. Consider `process.exit(0)` here since you've already determined the update needs user action, and the command is printed. Exit 1 implies error, not "needs manual intervention."

Alternatively, keep exit 1 but document why — scripts might pipe `ao update` and check `$?`.

File: packages/cli/src/commands/update.ts [hunk 10]
```diff
@@ -1,28 +1,167 @@
...
+  console.log(`Current version: ${chalk.dim(info.currentVersion)}`);
+  console.log(`Latest version:  ${chalk.green(info.latestVersion)}`);
+  console.log();
+
+  const command = "npm install -g @aoagents/ao@latest";
+
+  if (!isTTY()) {
+    // Non-interactive: print command but exit non-zero so scripts know no upgrade happened
+    console.log(`Run: ${chalk.cyan(command)}`);
+    process.exit(1);
```
Comment (i-trytoohard) [inline packages/cli/src/commands/update.ts]: ⚠️ **npm exit code 1 doesn't always mean permissions:** The sudo suggestion assumes permissions, but npm can exit 1 for many reasons (network, package not found, version conflict, engine mismatch). The suggestion `try: sudo npm install...` could mislead users into running sudo when that's not the issue.

Suggestion: Just print the exit code and the original command — let the user diagnose. Or check if the error message contains `EACCES` before suggesting sudo.

File: packages/cli/src/index.ts [hunk 11]
```diff
@@ -1,5 +1,14 @@
...
 
+import { maybeShowUpdateNotice, scheduleBackgroundRefresh } from "./lib/update-check.js";
+
+// Synchronous cache read — no network call on startup.
+maybeShowUpdateNotice();
+
+// Start background cache refresh early so it runs in parallel with the command.
+// The unref'd timer lets the process exit without waiting if the command finishes first.
+scheduleBackgroundRefresh();
+
```
Comment (i-trytoohard) [inline packages/cli/src/index.ts]: 💡 **Minor observation:** `scheduleBackgroundRefresh()` runs *before* `createProgram().parseAsync()`. This means the background fetch timer starts even for `--help`, `--version`, or invalid commands that exit immediately. Not a bug — the `unref()` handles it — but worth noting that help/version invocations trigger a registry fetch check that will be immediately cancelled.

File: packages/cli/src/lib/update-check.ts [hunk 12]
```diff
@@ -0,0 +1,320 @@
...
+ * Distinguishes global npm installs (e.g. /usr/local/lib/node_modules,
+ * ~/.nvm/.../lib/node_modules, pnpm global store) from local project
+ * node_modules by checking for `lib/node_modules` (global) vs a bare
+ * `node_modules` that sits inside a project directory (local/npx).
+ */
+export function classifyInstallPath(resolvedPath: string): InstallMethod {
+  const hasNodeModules =
+    resolvedPath.includes("/node_modules/") || resolvedPath.includes("\\node_modules\\");
+
+  if (hasNodeModules) {
```
Comment (i-trytoohard) [inline packages/cli/src/lib/update-check.ts:64]: 💡 **Edge case:** `classifyInstallPath` checks for `/lib/node_modules/` but doesn't account for Homebrew installs on macOS (`/opt/homebrew/lib/node_modules/...`). Homebrew npm uses `/opt/homebrew/lib/node_modules/` which *should* match the `lib/node_modules` heuristic. If anyone reports issues on macOS Homebrew, this is the place to check.

Not blocking — just a heads-up.

File: packages/cli/src/lib/update-check.ts [hunk 13]
```diff
@@ -0,0 +1,320 @@
...
+ * Falls back to the CLI package version if the wrapper is not resolvable
+ * (e.g. running from source where both are the same version anyway).
+ */
+export function getCurrentVersion(): string {
+  try {
+    const require = createRequire(import.meta.url);
+    const aoPkg = require("@aoagents/ao/package.json") as { version: string };
+    return aoPkg.version;
+  } catch {
+    return getCliVersion();
```
Comment (i-trytoohard) [inline packages/cli/src/lib/update-check.ts:123]: 💡 **Only 2 uncovered lines in the diff coverage report (99.3%):** Lines 117-118 — the `getCliVersion()` fallback in `getCurrentVersion()`. This triggers when `createRequire(import.meta.url)` can't resolve `@aoagents/ao/package.json`. Reasonable to skip since it's a defensive fallback for unusual environments, but if you want 100% you could mock `createRequire` to throw.

**Review by i-trytoohard — APPROVED** [review]
## Re-Review — PR #1156 (Round 2)

**Huge improvement.** All 12 issues from the first review have been addressed:

### ✅ Previously Flagged — All Fixed
1. ✅ **npm-global detection** — extracted `classifyInstallPath()` with 8 tests covering npm/nvm/Windows/pnpm-global/local-pnpm/npx/git/unknown
2. ✅ **npm update full flow** — 9 tests: confirm→spawn success, spawn failure, sudo suggestion, spawn ENOENT, non-TTY, flag warning, user decline, registry unreachable, forces fresh fetch
3. ✅ **maybeShowUpdateNotice happy path** — prints notice tested with version output
4. ✅ **AGENT_ORCHESTRATOR_CI** — tested
5. ✅ **Duplicate semver** — `isVersionOutdated` exported and reused in `doctor.ts`
6. ✅ **scheduleBackgroundRefresh** — 2 tests (no-throw, swallows errors)
7. ✅ **IS_TTY** — converted to `isTTY()` function, testable
8. ✅ **Registry unreachable misleading** — now exits with proper error message
9. ✅ **XDG_CACHE_HOME** — tested with override + fallback
10. ✅ **writeCache failures** — EACCES and ENOSPC tested, confirmed no throw
11. ✅ **--check with unreachable registry** — outputs valid JSON with null latestVersion
12. ✅ **isVersionOutdated edge cases** — 10 tests: major/minor/patch, equal, newer, missing parts, pre-release, NaN, same-with-prerelease

### 📊 Coverage Numbers
- **Diff coverage: 99.3%** (305 changed lines, only 2 uncovered — fallback in `getCurrentVersion`)
- **~105 tests total** across 4 test files (was 37 before)
- **All CI green:** typecheck, lint, tests, integration tests, web tests, diff coverage

### ⚠️ Minor (Non-Blocking)
- Non-TTY `process.exit(1)` — arguably should be exit 0 since it's informational
- Sudo suggestion on npm exit 1 assumes permissions when it could be anything
- Background refresh starts before parse, so even `--help` triggers it (unref cancels it though)

### Verdict
**Looks good. Ship it.** 🚀

**Review by i-trytoohard — APPROVED** [review]
## Re-Review — PR #1156 (Round 3)

**Both inline comments addressed cleanly:**

1. ✅ **Non-TTY now returns instead of `process.exit(1)`** — prints the command and exits cleanly (exit 0). Test updated: no longer expects `process.exit(1)`, just verifies the command is printed and prompt isn't called.

2. ✅ **Removed misleading sudo suggestion** — npm failure now just prints `npm exited with code ${code}` without assuming permissions. Test updated to check for `"exited with code 1"` instead of `"sudo"`.

Both changes are minimal, focused, and don't introduce new issues. Tests updated to match.

**All CI green. Diff coverage passing. Ready to merge.** 🚀

**Review by i-trytoohard — APPROVED** [review]
## Final Review — PR #1156

### Changes since last approval (1 commit)

Three targeted fixes, all clean:

1. **pnpm-global as first-class install method** — `InstallMethod` type now has 4 variants: `git | npm-global | pnpm-global | unknown`. `classifyInstallPath` checks pnpm global (`/pnpm/global/`) before npm global (`/lib/node_modules/`). `getUpdateCommand` returns `pnpm add -g` for pnpm installs. `update.ts` switch routes `pnpm-global` to `handleNpmUpdate` with `info.recommendedCommand` instead of hardcoded npm command. Tests updated.

2. **`scheduleBackgroundRefresh` moved to `.then()` after parseAsync** — no longer fires before the command runs. Short-lived commands (`--help`, `--version`) exit immediately without the timer holding the event loop open. Test mock updated from `.finally` to `.then`.

3. **`handleNpmUpdate` uses `info.recommendedCommand`** instead of hardcoded `"npm install -g ..."` — pnpm users get the right command.

### Final scorecard

| Check | Status |
|---|---|
| Typecheck | ✅ pass |
| Lint | ✅ pass |
| Tests | ✅ pass |
| Integration Tests | ✅ pass |
| Web Tests | ✅ pass |
| Diff Coverage | ✅ 99.4% (310 lines, 2 uncovered — fallback in `getCurrentVersion`) |
| NPM Audit | ✅ pass |
| Secrets Scan | ✅ pass |

### Files changed: 8 (+1728 / -40)

| File | Verdict |
|---|---|
| `src/lib/update-check.ts` | Clean. 4 install methods, pre-release handling, cache with 24h TTL. |
| `src/commands/update.ts` | Clean. Install-aware routing, proper error messages, non-TTY safe. |
| `src/commands/doctor.ts` | Clean. Reuses `isVersionOutdated`, no duplication. |
| `src/index.ts` | Clean. Notifier → parse → background refresh. |
| `__tests__/lib/update-check.test.ts` | 781 lines, ~70 tests. Thorough. |
| `__tests__/commands/update.test.ts` | ~24 tests covering all 4 install methods + edge cases. |
| `__tests__/commands/doctor.test.ts` | 3 tests for version freshness. |
| `__tests__/index.test.ts` | 3 tests including error handling. |

**No open issues. Ship it.** 🚀
